File: conting/PIPELINE/fluxos.f
c23456781234567812345678123456781234567812345678123456781234567812345678
c   1       2       3       4       5       6       7       8       9
c ----------------------------------------------------------------------
c
c ----------------------------------------------------------------------
c Subrotina para calcular os fluxos de potencia nos ramos do sistema
c  <<< ATIVO >>>   <<< REATIVO >>>  <<< ATIVO >>>   <<< REATIVO >>>

      subroutine fluxos    ! chamada pelo programa principal
c
      
      include 'globvar.inc'
      include 'varcont.inc'

      intrinsic sin,cos
   
      do 30 jj=1,nl
        k=ni(jj)
        m=nf(jj)
        tkm=t(k)-t(m)
        g=gkl(jj)
        b=bkl(jj)
        co=cos(tkm)
        si=sin(tkm)
        vk=v(k)
        vkk=vk*vk
        vm=v(m)
        vmm=vm*vm 
        ta=tap(jj)
        ysh=ylin(jj)
        if(ta .eq. 0.0) then
          t1=1.0
          goto 90
        endif
        t1=1.0/ta
  90    bb=b+ysh        

        fpkm(jj)=    t1*g*vkk - vk*vm*(g*co+b*si)    ! fluxo ativo
        rqkm(jj)=  -t1*bb*vkk - vk*vm*(g*si-b*co)    ! fluxo reativo
        fpmk(jj)=  (g/t1)*vmm - vk*vm*(g*co-b*si)     
        rqmk(jj)=-(bb/t1)*vmm - vk*vm*(-g*si-b*co)

        fpjk(jj) = fpkm(jj)

  30  continue

      return
      end         ! Fim da subrotina de calculo de fluxos ativos e reativos

c ----------------------------------------------------------------------

c Subrotina para calcular fluxos de potencia ativa ramos do sistema
c   <<< ATIVO >>>    <<< ATIVO >>>   <<< ATIVO >>>   <<< ATIVO >>>

c     [  Somente para analise de contingencias  ]

      subroutine fluxativo     
c
      include 'globvar.inc'
      include 'varcont.inc'
      intrinsic sin,cos
   
      
      do 30 jj=1,nl
        k=ni(jj)
        m=nf(jj)
        tkm=t(k)-t(m)
        g=gkl(jj)
        b=bkl(jj)
        co=cos(tkm)
        si=sin(tkm)
        vk=v(k)
        vkk=vk*vk
        vm=v(m)
        vmm=vm*vm 
        ta=tap(jj)
        if(ta .eq. 0.0) then
          t1=1.0
          goto 90
        endif
        t1=1.0/ta
  90    continue        

        fpjk(jj)=    t1*g*vkk - vk*vm*(g*co+b*si)    ! fluxo ativo
cc        fpkj(jj)=  (g/t1)*vmm - vk*vm*(g*co-b*si) 

cc        print*,'Em fluxativo() --> barra ',k,' a barra ',m,' ramo ',jj
cc        print*,'Fluxo CB --> fpjk(',jj,') = ',fpjk(jj)

  30  continue

      return
      end         ! Fim da subrotina de calculo de fluxos ativos

File: estagio/conting/conting/fluxos.c
/*------------------------------fluxos.c---------------------------------------------*/

#include <stdlib.h>
#include <math.h>
#include "global.h"

/*-------------------Inicio da funcao fluxos-------------------------------*/
/* Funcao para calcular fluxos de potencia ativa e reativo nos ramos do sistema*/
/*-------------------------- MODELO AC DA REDE ELETRICA--------------------------*/

void fluxos(void){
	int k,m,j;
	double tkm,vkk,vkm,vmm,bb,co,si,t1;
	
	for (j=0; j<nl; j++){
		k = ni[j];
		m = nf[j];
		tkm = t[k]-t[m];
		co = cos(tkm);
		si = sin(tkm);
		vkk = v[k]*v[k];
		vmm = v[m]*v[m];
		vkm = v[k]*v[m];
		if( tap[j] == 0.e0 )
			t1 = 1.e0;
		else
			t1 = 1.e0/tap[j];
		
		bb = bkl[j]+ylin[j];
		fpkm[j] = t1*gkl[j] *vkk - vkm*(gkl[j]*co+bkl[j]*si);    // fluxo ativo
		rqkm[j] =  -t1* bb  *vkk - vkm*(gkl[j]*si-bkl[j]*co);    // fluxo reativo
		fpmk[j] =(gkl[j]/t1)*vmm - vkm*(gkl[j]*co-bkl[j]*si);
		rqmk[j] = -( bb /t1 )*vmm - vkm*(-gkl[j]*si-bkl[j]*co);
		fpjk[j] = fpkm[j];
	}
	return;
}  
/*----------------------Fim da funcao fluxos-----------------------------*/

/* Funcao para calcular fluxos de potencia ativa e reativa dos ramos do sistema   
        [ Somente prar Screening de Contingencia ]     */

void fluxpq(void){
	int k,m,jj;
	double tkm,vkk,vkm,vmm,bb,co,si,t1;
	for (jj=0; jj<nl; jj++){
		k = ni[jj];
		m = nf[jj];
		tkm = t[k]-t[m];
		co = cos(tkm);
		si = sin(tkm);
		vkk = v[k]*v[k];
		vmm = v[m]*v[m];
		vkm = v[k]*v[m];
		bb = bkl[jj]+ylin[jj];
		if( tap[jj] == 0.e0 )
			t1 = 1.e0;
		else
			t1 = 1.e0/tap[jj];
		fpjk[jj] = t1*gkl[jj]*vkk - vkm*(gkl[jj]*co + bkl[jj]*si); // fluxo ativo
		fqjk[jj] = -t1*bb*vkk - vkm *(gkl[jj]*si - bkl[jj]*co); // fluxo reativo
	}
	return;
}
/*------------Fim da funcao fluxpq------------------*/
